poh/src/poh_recorder.rs

```diff
@@ -1006,6 +1024,57 @@ impl SharedWorkingBank {
     }
 }
 
+/// Wrapper around a atomic-u64 that prevents modifying outside
+/// of `PohRecorder`.
+#[derive(Clone)]
+pub struct SharedTickHeight(Arc<AtomicU64>);
+
+impl SharedTickHeight {
+    pub fn load(&self) -> u64 {
+        self.0.load(Ordering::Acquire)
+    }
+
+    fn new(tick_height: u64) -> Self {
+        Self(Arc::new(AtomicU64::new(tick_height)))
+    }
+
+    fn store(&self, tick_height: u64) {
+        self.0.store(tick_height, Ordering::Release);
+    }
+
+    fn increment(&self) {
+        self.0.fetch_add(1, Ordering::Release);
+    }
```

Comment:
Private methods that mutate the inner atomic should take a mutable self so that we guarantee that these methods are only callable when poh recorder is write locked. This comment applies to `SharedLeaderFirstTickHeight` as well as `SharedWorkingBank`